internal/store/authprovider.go

```diff
@@ -0,0 +1,158 @@
+package store
+
+import (
+	"context"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type AuthProvider interface {
+	InitialMigration(ctx context.Context) error
+
+	Create(ctx context.Context, orgId uuid.UUID, authProvider *api.AuthProvider, eventCallback EventCallback) (*api.AuthProvider, error)
```

Comment:
question: does making the auth provider org scoped cause a circular dependency? caller needs an orgId to find  the provider to authenticate to then determine orgId.